RF24Mesh.cpp

```diff
@@ -314,49 +314,47 @@ bool ESBMesh<network_t, radio_t>::requestAddress(uint8_t level)
 {
     RF24NetworkHeader header(MESH_MULTICAST_ADDRESS, NETWORK_POLL);
     //Find another radio, starting with level 0 multicast
-    IF_MESH_DEBUG(printf_P(PSTR("%u: MSH Poll\n"), millis()));
+    IF_MESH_DEBUG(printf_P(PSTR("MSH Poll Level %d\n"), level));
     network.multicast(header, 0, 0, level);
 
-    uint32_t timr = millis();
+    uint32_t pollTimeout = millis() + 55;
 #define MESH_MAXPOLLS 4
     uint16_t contactNode[MESH_MAXPOLLS];
     uint8_t pollCount = 0;
 
-    while (1) {
+    while (millis() < pollTimeout && pollCount < MESH_MAXPOLLS) {
 #if defined(MESH_DEBUG)
         bool goodSignal = radio.testRPD();
 #endif
-
         if (network.update() == NETWORK_POLL) {
-
-            memcpy(&contactNode[pollCount], &network.frame_buffer[0], sizeof(uint16_t));
-            if (pollCount > 0 && contactNode[pollCount] != contactNode[pollCount - 1]) { //Drop duplicate polls to help prevent duplicate requests
-                ++pollCount;
+            uint16_t contact = 0;
+            memcpy(&contact, &network.frame_buffer[0], sizeof(contact));
+
+            // Drop duplicate polls to help prevent duplicate requests
+            bool isDupe = false;
+            for (uint8_t i = 0; i < pollCount; ++i) {
+                if (contact == contactNode[i]) {
+                    isDupe = true;
+                    break;
+                }
             }
-            else {
+            if (!isDupe) {
+                contactNode[pollCount] = contact;
                 ++pollCount;
+                IF_MESH_DEBUG(printf_P(PSTR("MSH Poll %c -64dbm from 0%o \n"), (goodSignal ? '>' : '<'), contact));
             }
 
-            IF_MESH_DEBUG(printf_P(PSTR("%u: MSH Poll %c -64dbm\n"), millis(), (goodSignal ? '>' : '<')));
         } // end if
 
-        if (millis() - timr > 55 || pollCount >= MESH_MAXPOLLS) {
-            if (!pollCount) {
-                IF_MESH_DEBUG(printf_P(PSTR("%u: MSH No poll from level %d\n"), millis(), level));
-                return 0;
-            }
```

Comment:
Because this was removed. We might want to add a `if (!pollCount) return 0;` just before the `for` loop below (but after the debug statement showing the poll count). This is to avoid allocating uint8_t for iterating an empty array of `contactNode`s.

Reply:
This actually reduces the code size by 2-bytes on a Nano, done!